documentation.md

```diff
@@ -1,154 +1,197 @@
 # BigUnit
 
 ## Overview
+
 The `BigUnit` package is a TypeScript library designed to manage numbers of varying precision, such as cryptocurrencies.
 
 # BigUnitish Type
+
 Represents a type that can be converted to a `BigUnit` if a precision value is available. It can be a number, string, bigint, or `BigUnit` itself.
 
 # BigUnit Class
 
 ## Methods
 
-Arithmetic operations return another BigUnit instance with the precision of the highest of the two numbers. To change to another precision, append `asPrecision(myPrecision)`. 
-If the parameter passed into an arithmetic function is not an instance of `BigUnit`, it will be converted to a `BigUnit` via `BigUnit.from()`. For example, assuming `myUnit` is a `BigUnit` instance with a precision of 5,
-`myUnit.add(2)` will add `2` units to myBigUnit. The exception to this is when the parameter is a `bigint`. In this case, the precision must be passed as a parameter.
-This improves clarity and prevents errors when working with different precisions.
+If the parameter passed into an arithmetic function is not an instance of `BigUnit`, it will be converted to a `BigUnit` via `BigUnit.from()` using the precision of the parent unless a precision is passed as a parameter. For example, assuming `myValue` is a `BigUnit` instance with a precision of 2, `myValue.add(0.12345)` will add `0.12` units to myValue whereas `myValue.add(0.12345, 5)` will add `0.12345` units to myValue. The exception to this is when the parameter is a `bigint`, in which case, the precision must be passed as a parameter. This improves clarity and prevents errors when working with different precisions. Arithmetic operations return another BigUnit instance with the precision of the highest of the two numbers. To convert the output to another precision, append `.asPrecision(myPrecision)`.
```

Comment:
While this is factually accurate, hitting the reader with this as the very first paragraph is possibly not a good idea. I think it's better that this level of the documentation is a surface-level summary. Deep technical details should be explained in a comprehensive technical document.

I think a more comprehensive technical document is a good idea, and this text belongs in it.